lua/entities/gmod_wire_expression2/core/timer.lua

```diff
@@ -113,10 +200,178 @@ e2function void stopAllTimers()
+	luaTimerCreate(self, name, delay, 1, callback)
+end
+
+__e2setcost(5)
+e2function void timerSetDelay(string name, number delay)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].delay = delay
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), delay, 0)
+end
+
+e2function number timerSetReps(string name, number repetitions)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].repetitions = repetitions
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), luaTimers[entIndex][name].delay, repetitions)
+end
+
+e2function void timerAdjust(string name, number delay, number repetitions)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].delay = delay
+	luaTimers[entIndex][name].repetitions = repetitions
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), delay, repetitions)
+end
+
+-- __e2setcost(20)
+-- e2function void timerSetCallback(string name, function callback)
+-- 	if not luaTimerExists(self, name) then
+-- 		return self:throw("Timer with name " .. name .. " does not exist", nil)
+-- 	end
+
+-- 	local int_name = luaTimerGetInternalName(self.entity:EntIndex(), name)
+-- 	local luaTimer = table.Copy(luaTimers[self.entity:EntIndex()][int_name])
+-- 	luaTimerRemove(self, name)
+
+-- 	luaTimer['callback'] = callback
+-- 	luaTimerCreate(self, name, luaTimer.delay, luaTimer.repetitions, callback)
+-- end
+
+__e2setcost(1)
+[nodiscard]
+e2function number timerGetDelay(string name)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", 0)
+	end
+
+	return luaTimers[self.entity:EntIndex()][name].delay
+end
+
+[nodiscard]
+e2function number timerGetReps(string name)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", 0)
+	end
+
+	return luaTimers[self.entity:EntIndex()][name].repetitions
+end
+
+[nodiscard]
+e2function function timerGetCallback(string name)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", "")
+	end
+
+	return luaTimers[self.entity:EntIndex()][name].callback
+end
+
+__e2setcost(5)
+e2function void timerRestart(string name)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	local internalName = luaTimerGetInternalName(entIndex, name)
+	timer.Stop(internalName)
+	timer.Adjust(name, luaTimers[entIndex][name].delay, luaTimers[entIndex][name].repetitions)
+	timer.Start(internalName)
```

Comment:
Is the stop and start really needed?

Reply:
Also, should 'name' be 'internalName' in line 317?

Reply:
Fixes